keps/sig-node/1592-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1206 @@
+
+
+## Motivation
+
+
+Kubernetes workloads often consist of multiple containers collaborating within a pod. While this separation enables granular monitoring and management of each container, it can be challenging to accurately estimate and allocate resources for individual containers, especially for workloads with unpredictable or bursty resource demands. This often leads to over-allocation of resources to ensure that no container experiences resource starvation, as kubernetes currently lacks a mechanism for sharing resources between containers easily. 
+
+Introducing pod-level resource requests and limits offers a more simplified resource management for multi-container pods as it is easier to gauge the collective resource usage of all containers in a pod rather than predicting each container’s individual needs. This proposal aims to support pod-level resource requests and limits that would allow Kubernetes to manage the total resource consumption of the pod, freeing users from the burden of meticulously allocating resources for each container. This simplified approach is particularly valuable for tightly coupled applications, and it will enable Kubernetes to leverage the underlying cgroup resource management mechanisms more effectively for better resource utilization for bursty workloads. 
+
+This new feature of resource specification at pod-level would complement, not replace, the existing container-level limits, providing users with greater flexibility.
+
+
+### Goals
+
+This proposal aims to:
+
+1. Extend Pod API to allow specifying resource limits on pod level in addition to individual container level. 
+2. Support Pod-level resource requests/limits as an opt-in feature that is compatible with existing usage of container level resource/limits.
+3. Use as much as possible the Linux kernel’s cgroup capabilities for resource management between containers within a pod’s cgroup hierarchy.
+
+
+### Non Goals
+
+
+1. This KEP focuses on the core resource types of CPU and memory. It doesn’t intend to address other resource types (e.g. GPU, storage, network bandwidth) at pod level in this phase. However, it could be considered in future extensions of the KEP.
+
+
+## Proposal
+
+
+### User Stories
+
+
+#### Story 1
+
+A development environment implemented as a Kubernetes Pod allows for separation of tools and a (web-based) IDE between multiple side-cars. The development environment defines a container with the web-server serving the IDE itself, and constrains it to use a certain amount of memory. Additional tools are provided in additional side-car deployments - for example an LSP( https://langserver.org/) service, a terminal, and more. Using this new feature, the containers providing the terminal, the LSP services, and the set of tools being utilized can share the resource limit defined for the pod. Consider the following pod definition:
+
+```yaml
+
+apiVersion: v1
+
```

Comment:
nit: could you drop the empty lines in between? I think that it will make it easier to read.

Reply:
done, the tool I used to convert doc to readme messed up the formatting.